pkg/provisioner/ironic/ironic.go

```diff
@@ -1155,6 +1155,93 @@ func (p *ironicProvisioner) ironicHasSameImage(ironicNode *nodes.Node) (sameImag
 	return sameImage
 }
 
+func (p *ironicProvisioner) buildManualCleaningSteps() (cleanSteps []nodes.CleanStep) {
+	// TODO: Add manual cleaning steps for host configuration
+
+	return
+}
+
+func (p *ironicProvisioner) startManualCleaning(ironicNode *nodes.Node) (success bool, result provisioner.Result, err error) {
+	cleanSteps := p.buildManualCleaningSteps()
+
+	// Start manual clean
+	if len(cleanSteps) != 0 {
+		p.log.Info("remove existing configuration and set new configuration", "steps", cleanSteps)
+		return p.tryChangeNodeProvisionState(
+			ironicNode,
+			nodes.ProvisionStateOpts{
+				Target:     nodes.TargetClean,
+				CleanSteps: cleanSteps,
+			},
+		)
+	}
+	result, err = operationComplete()
+	return
+}
+
+// Prepare remove existing configuration and set new configuration.
+// If `started` is true,  it means that we successfully executed `tryChangeNodeProvisionState`.
+func (p *ironicProvisioner) Prepare(unprepared bool) (result provisioner.Result, started bool, err error) {
```

Comment:
What about covering this part using the new mocks? I think that for such relevant feature could be really useful